AIPscan/Reporter/views.py

```diff
@@ -86,25 +86,36 @@ def storage_locations_with_aips(storage_locations):
     return [loc for loc in storage_locations if loc.aips]
 
 
-def get_aip_pager(page, per_page, storage_service, storage_location):
-    try:
-        page = int(page)
-    except ValueError:
-        page = 1
-
+def get_aip_pager(page, per_page, storage_service, **kwargs):
     storage_service_id = None
     if storage_service is not None:
         storage_service_id = storage_service.id
 
-    pager = AIP.query.filter_by(storage_service_id=storage_service_id).paginate(
-        page=page, per_page=per_page, error_out=False
-    )
+    storage_location_id = None
+    if kwargs.get("storage_location", None):
+        storage_location_id = kwargs["storage_location"].id
 
-    if storage_location:
-        pager = pager.query.filter_by(storage_location_id=storage_location.id).paginate(
-            page=page, per_page=per_page, error_out=False
+    query = kwargs.get("query", None)
+
+    aips = AIP.query
+
+    # Filter by storage service
+    aips = aips.filter_by(storage_service_id=storage_service_id)
+
+    # Optionally filter by storage location
+    if storage_location_id:
+        aips = aips.filter_by(storage_location_id=storage_location_id)
+
+    # Optionally filter by text search
+    if query is not None:
+        aips = aips.filter(
+            AIP.uuid.like(f"%{query}%")
+            | AIP.transfer_name.like(f"%{query}%")
+            | AIP.create_date.like(f"%{query}%")
         )
```

Comment:
This guarantees a full table scan, it won't be very efficient. I guess that's ok for now?

Reply:
Yeah, this is just to get something in place to replace functionality that seemingly once existed (a client had made their own documentation of it with screenshots), yet weirdly doesn't seem to exist in the code (I scoured the commit history and could find any trade). Very weird!

If it's slow for them I might re-implement it with Typesense.